programs/platform.h

```diff
@@ -0,0 +1,148 @@
+/*
+    platform.h - compiler and OS detection
+    Copyright (C) 2016-present, Przemyslaw Skibinski, Yann Collet
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#ifndef PLATFORM_H_MODULE
+#define PLATFORM_H_MODULE
+
+#if defined (__cplusplus)
+extern "C" {
+#endif
+
+
+/* **************************************
+*  Compiler Options
+****************************************/
+#if defined(__INTEL_COMPILER)
+#  pragma warning(disable : 177)    /* disable: message #177: function was declared but never referenced */
```

Comment:
OK, here is one typical issue with defining such compilation flag within `platform.h` : 
now, non-referenced function warning is disabled across a large portion of source code (at least, any source which includes directly or indirectly `platform.h`).

That could be a good idea if there was some functions within  `platform.h` that could uselessly trigger it, but that's not the case here : I don't see any function declaration in `platform.h`, only macros, so the flag is enabled _on behalf of future users_, such as, maybe, `util.h`.

But now, a user which only includes `platform.h` has a useful warning flag disabled. This flag could help him detect bugs before they get more complex to understand.

Disabling a warning flag can be completely legitimate, but it's better done close the section where it's needed, and better avoided as a "global setting".

If the point here is to disable it for `util.h`, better do it within `util.h` (and even better, re-enable it at the end of the file). I suspect it's related to `UTIL_STATIC`, in which case this is the better place to define it.